model/tests/test_util.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+
+from model.util import SpikeMovingAverage, TemporalFilter, VarianceMovingAverage
+
+
+def test_temporal_filter() -> None:
+    tf = TemporalFilter(tau_rise=1, tau_fall=1)
+
+    # Initial apply since temporal filter is second order
+    tf.apply(error=1)
+
+    # Apply a single error with a small dt should result in small change
+    assert tf.apply(error=1) == pytest.approx(1.7357588823428847)
```

Comment:
Would this be more intuitive if written in terms of math.exp()?

Reply:
Yes I think that this is a smell that it isn't more intuitive.

The result of `tf.apply()` will be resultant from that iterative update of `tf.rise` and `tf.fall` class vars.  I couldn't think of a way to do this more cleanly. We could pull out the logic from the implementation of `tf.apply()`, but doing it for multiple timesteps of `apply()` would end up with a huge chain of operations. 

So in that case I'd rather have the number. The thought being: looks like it works now, this code probably won't change much once we get the filter ironed out so this number smell will cause minimal disruption.

If you think I am missing something lmk!